### PR TITLE
Domains: Poperly handle 'empty_results' error from the backend

### PR DESCRIPTION
**Pull request by klimeryk**
In some rare cases, it's possible we won't be able to generate any suggestions for a query. It might be because many (or all) TLDs are in maintenance mode, for example. In those cases, the backend returns an error, `empty_results`. We need to handle that case explicitly - otherwise, a generic, confusing error is shown to the user.

### Testing
Before:
screen shot 2017-12-03 at 05 23 50

After:
no error is shown ;)

How to test:
Best to hack the backend to return a `WP_Error` when needed:
* Return `empty_result` for all calls to `/suggestions` - no errors should be shown on the frontend. It's a super-edge case that should not happen, but 🤷‍♂️ 
* Return `empty_result` only for call to domainsbot, not the wpcom subdomain suggestion.
* Return `empty_result` for the wpcom subdomain suggestion - the rest of the suggestions should be displayed, but the subdomain suggestion should not remain in the loading state (this was broken before this PR).
* Same as previous point, but a different error (very rare, but technically can happen). Should display the error, wpcom subdomain suggestion should not be in loading state.

### CODE REVIEW

**Comment by aidvu** [comment]
@klimeryk breaks domain search in `Domain -> Add`.

screen shot 2017-12-04 at 15 34 59

**Comment by klimeryk** [comment]
Thanks for the review, @aidvu. But I could not reproduce the error you were seeing - any special steps I need to take? However, I did notice a different error - seems that there were two places that we were not handling errors from the backend properly (we were passing in a `{ code: 'empty_results' }` object to the `getAvailabilityNotice` function, instead of just the raw string. This resulted in the generic `Sorry, there was a problem processing your request. Please try again in a few minutes.` error in some situations. I'll update the testing instructions with those cases.

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -579,6 +575,11 @@ class RegisterDomainStep extends React.Component {
 						-1,
 						this.props.analyticsSection
 					);
+
+					this.setState( {
```
Comment (klimeryk) [inline client/components/domains/register-domain-step/index.jsx]: Without this, the wpcom subdomain suggestion would remain in the loading state when an error occurs. Of course, it's super rare for that to happen, but 🤷‍♂️

**Comment by aidvu** [comment]
Will recheck today.

**Comment by aidvu** [comment]
Yep, got it right the first time. If there is nothing in `localStorage` and we return `empty_result`, it breaks domain search.

Check: https://github.com/Automattic/wp-calypso/blob/master/client/components/domains/domain-search-results/index.jsx#L194

Steps:
- Make the back-end always return `empty_result` (which I guess is what happened)
- Open an incognito window
- Go to `Domains -> Add`
- See placeholders blinking endlessly 😭 
- Try searching
- 💥

**Comment by aidvu** [comment]
To make it clear, this wasn't introduced by you, it will break on regular Calypso too, but I'd rather fix it to work with empty `localStorage` when we're at it.

**Comment by klimeryk** [comment]
Thanks, @aidvu, nice catch! I was able to reproduce it and improve the error handling even more. Of course, it's a super edge case, but we should handle it nevertheless.
I agree that if there are no results, we should have just returned an empty array. But I'd prefer to do that in a separate patch - it seems we are assuming "empty suggestions = still loading". And it'd require a new version of the endpoint... so, not looking forward to that - this change has been a PITA enough as it was 😅

File: client/components/domains/register-domain-step/index.jsx (hunk 1 above)
Comment (delputnam) [reply client/components/domains/register-domain-step/index.jsx]: Nice catch. Thank you.
